packages/cardhost/app/components/cardhost-left-edge.js

```diff
@@ -0,0 +1,59 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+export default class CardhostTopEdgeComponent extends Component {
+  @service cardstackSession;
+  @tracked isExpanded = false;
+
+  @action
+  toggleMenuExpand() {
+    this.isExpanded = !this.isExpanded;
+  }
+
+  @service router;
+
+  @action
+  logout(sessionLogout) {
+    sessionLogout();
+    let cardId = this.router.currentRoute.parent.params.name;
+    if (cardId) {
+      this.router.transitionTo('cards.card.view', cardId);
+    } else {
+      this.router.transitionTo('index');
+    }
+  }
+
+  @action
+  closeLeftEdge() {
+    this.isExpanded = false;
+  }
+
+  /*
+    closeListener, registerCloseListener, and destroyCloseListener are the
+    actions responsible for closing the left edge when you click somewhere
+    outside of it. They are only registered when the left edge is open,
+    and torn down when it is closed via the did-insert and will-destroy
+    render modifiers.
+  */
+  @action
```

Comment:
you could simplify this by putting a `{{did-update this.toggleListener this.isExpanded}}` on the `#cardhost-left-edge` element, and do something like
```
  @action
  closeListener(element) {
    return event => {
      if (!element.contains(event.target)) {
        this.closeLeftEdge();
      }
    };
  }

  @action
  toggleListener(el, [expanded]) {
    let listener = expanded ? 'removeEventListener' : 'addEventListener';
    ['click', 'focusIn'].forEach(evt => document.querySelector('body')[listener](evt, this.closeListener(el)));
  }
```

Reply:
although I suppose the argument could be made to keep it more explicit 🤷‍♂

Reply:
I think we need to keep it as `will-destroy` because we need to ensure that we tear down this click handler when we leave the route as well.